src/reanimated2/initializers.ts

```diff
@@ -0,0 +1,150 @@
+import NativeReanimatedModule from './NativeReanimated';
+import { runOnUI, runOnJS } from './threads';
+
+function callGuard<T extends Array<any>, U>(
+  fn: (...args: T) => U,
+  ...args: T
+): void {
+  'worklet';
+  try {
+    fn(...args);
+  } catch (e) {
+    if (global.ErrorUtils) {
+      global.ErrorUtils.reportFatalError(e);
+    } else {
+      throw e;
+    }
+  }
+}
+
+function valueUnpacker(objectToUnpack: any, category?: string): any {
+  'worklet';
+  let workletsCache = global.__workletsCache;
+  let handleCache = global.__handleCache;
+  if (workletsCache === undefined) {
+    // init
+    workletsCache = global.__workletsCache = new Map();
+    handleCache = global.__handleCache = new WeakMap();
+  }
+  if (objectToUnpack.__workletHash) {
+    let workletFun = workletsCache.get(objectToUnpack.__workletHash);
+    if (workletFun === undefined) {
+      const evalFn =
+        global.evalWithSourceMap || global.evalWithSourceUrl || eval; // eslint-disable-line no-eval
+      workletFun = evalFn(
+        '(' + objectToUnpack.asString + '\n)',
+        `worklet_${objectToUnpack.__workletHash}`,
+        objectToUnpack.__sourceMap
+      ) as (...args: any[]) => any;
+      workletsCache.set(objectToUnpack.__workletHash, workletFun);
+    }
+    const functionInstance = workletFun.bind(objectToUnpack);
+    objectToUnpack._recur = functionInstance;
+    return functionInstance;
+  } else if (objectToUnpack.__init) {
+    let value = handleCache!.get(objectToUnpack);
+    if (value === undefined) {
+      value = objectToUnpack.__init();
+      handleCache!.set(objectToUnpack, value);
+    }
+    return value;
+  } else if (category === 'RemoteFunction') {
+    const fun = () => {
+      throw new Error(`Tried to synchronously call a non-worklet function on the UI thread.
+
+Possible solutions are:
+  a) If you want to synchronously execute this method, mark it as a worklet
+  b) If you want to execute this function on the JS thread, wrap it using \`runOnJS\``);
+    };
+    fun.__remoteFunction = objectToUnpack;
+    return fun;
+  } else {
+    throw new Error('data type not recognized by unpack method');
+  }
+}
+
+type StackDetails = [Error, number, number];
+
+const _workletStackDetails = new Map<number, StackDetails>();
+
+export function registerWorkletStackDetails(
+  hash: number,
+  stackDetails: StackDetails
+) {
+  _workletStackDetails.set(hash, stackDetails);
+}
+
+function getBundleOffset(error: Error): [string, number, number] {
+  const frame = error.stack?.split('\n')?.[0];
+  if (frame) {
+    const parsedFrame = /@(.*):(\d+):(\d+)/.exec(frame);
```

Comment:
We will need to check these regular expressions for vulnerabilities (some time ago we had one in `Colors.ts`, see https://github.com/advisories/GHSA-2j79-8pqc-r7x6).

Reply:
wouldn't really be concerned about this, there are perhaps ways of making this regex run forever but that would've need to be a user code that triggers this behavior. If the user pulls in a malicious code into their app there are definitely other, easier to exploit vectors of attack than making some regex that runs in development builds only to execute in infinite loop